localstack/aws/skeleton.py

```diff
@@ -22,6 +24,19 @@
 DispatchTable = Dict[str, ServiceRequestHandler]
 
 
+# Cache the client to avoid instantiation per bound method. Also, they are thread-safe with some
+# caveats
+# https://boto3.amazonaws.com/v1/documentation/api/latest/guide/clients.html
+@lru_cache
+def boto_client(service_name):
+    return boto3.client(service_name)
+
+
```

Comment:
boto setup to bound a function to `ServiceRequestDispatcher` that is not the provider implementation (non-override use case)